### ISSUE TITLE
Return null as JSX element (no-null with React + TS)

### ISSUE THREAD

**Issue by tiagoporto** [sindresorhus/eslint-plugin-unicorn#964]
Is necessary return null for a JSX Element

![Screen Shot 2020-12-26 at 8 31 25 PM](https://user-images.githubusercontent.com/3688905/103160918-6beabd80-47b9-11eb-9160-87ff5a05f379.png)

If use **undefined** TS return an error

```
Its return type 'Element | undefined' is not a valid JSX element.
Type 'undefined' is not assignable to type 'Element | null'.
```

Maybe enable null for a JSX return fix the bug.

**Comment by sindresorhus** [comment]
We could maybe ignore `null` when in a JSX context or something.

**Comment by Torhal** [comment]
Or possibly return an empty React.Fragment. I do that instead of using null.

```js
return isLoading
    ? <></> 
    : (
            /* ... */
       )
```

**Comment by sindresorhus** [comment]
How about when in a JSX context, we change the message to recommend using `<></>` instead of `null`?

**Comment by tiagoporto** [comment]
Fragments fix the issue, but I personally think the best way is using `null`.

As you can check on this last example of this post [Use ternaries rather than && in JSX](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx), Kent C Dodds recommends `null`.

Since nothing should be rendered, it's not necessary to do any other processing.

And even not rendering any element the [Fragments](https://reactjs.org/docs/fragments.html)' purpose is to render a group of elements.

> A common pattern in React is for a component to return multiple elements. Fragments let you group a list of children without adding extra nodes to the DOM.

**Comment by Torhal** [comment]
> Fragments fix the issue, but I personally think the best way is using `null`.

Since I'm using the ```unicorn/no-null``` rule, I'd rather use fragments without children.

**Comment by tiagoporto** [comment]
@Torhal I see your point, but return null is the recommendation even in react docs [Preventing Component from Rendering](https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering).

Add `null` on this case it's not wrong, but necessary.

This rule could check if the return is a JSX to enable `null`.

**Comment by sindresorhus** [comment]
React 18 will [allow returning `undefined`](https://github.com/reactwg/react-18/discussions/75), so I'm gonna close this as there is a known workaround in the meantime.

**Comment by SStranks** [comment]
> React 18 will [allow returning `undefined`](https://github.com/reactwg/react-18/discussions/75), so I'm gonna close this as there is a known workaround in the meantime.

How does this work in practice though? I still get the same error as the OP if I try returning undefined when it is expecting JSX.

**Comment by yakov116** [comment]
@SStranks commenting on an issue that was closed 2 years ago is never a good idea.

The proper thing to do is open a new issue.
